src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockRawlsDAO.scala

```diff
@@ -144,15 +144,15 @@ class MockRawlsDAO  extends RawlsDAO {
   override def adminAddMemberToGroup(groupName: String, memberList: RawlsGroupMemberList): Future[Boolean] = {
     val userEmailsToAdd = memberList.userSubjectIds.getOrElse(Seq[String]()).toSet
     val groupWithNewMembers = (groupName -> ((groups(groupName).filterNot(userEmailsToAdd.contains)) ++ userEmailsToAdd))
-    groups = groups + groupWithNewMembers
+    groups = groups.synchronized { groups + groupWithNewMembers }
```

Comment:
But the assignment to `groups` still happens outside of the synchronized block? This would synchronize both reads and writes to `groups`:
```
groups.synchronized { groups = groups + groupWithNewMembers }
```
Or even:
```
this.synchronized { groups = groups + groupWithNewMembers }
```